From 370a20576c9d9e1b49f25b7f323b8c20013ce043 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Tue, 26 Mar 2019 14:18:30 +0530
Subject: [PATCH 0106/1921] octeontx2-af: Increase register poll timeout

NIX calibration is taking fairly longtime to complete.
Hence increasing the register polling timeout.

Change-Id: I143f3d1552d7933b36266ff80c526276af8d40df
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 5a23041369af..1592e63dc4d9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -61,7 +61,7 @@ MODULE_PARM_DESC(mkex_profile, "MKEX profile name string");
  */
 int rvu_poll_reg(struct rvu *rvu, u64 block, u64 offset, u64 mask, bool zero)
 {
-	unsigned long timeout = jiffies + usecs_to_jiffies(100);
+	unsigned long timeout = jiffies + usecs_to_jiffies(10000);
 	void __iomem *reg;
 	u64 reg_val;
 
@@ -73,7 +73,6 @@ int rvu_poll_reg(struct rvu *rvu, u64 block, u64 offset, u64 mask, bool zero)
 		if (!zero && (reg_val & mask))
 			return 0;
 		usleep_range(1, 5);
-		timeout--;
 	}
 	return -EBUSY;
 }
-- 
2.31.1

